bootstore/src/schemes/v0/peer_networking.rs

```diff
@@ -0,0 +1,635 @@
+        Err(ConnErr::Retry)
+    }
+}
+/// Spawn a task that maintains a client connection to a peer
+pub async fn spawn_client(
+    unique_id: u64,
+    my_peer_id: Baseboard,
+    my_addr: SocketAddrV6,
+    addr: SocketAddrV6,
+    log: &Logger,
+    main_tx: mpsc::Sender<ConnToMainMsg>,
+) -> PeerConnHandle {
+    // Create a channel for sending `MainToConnMsg`s to this connection task
+    let (tx, mut rx) = mpsc::channel(2);
+    let log = log.clone();
+
+    let handle = tokio::spawn(async move {
+        loop {
+            let sock = match TcpStream::connect(addr).await {
+                Ok(sock) => sock,
+                Err(err) => {
+                    // TODO: Throttle this?
+                    warn!(log, "Failed to connect"; "addr" => addr.to_string());
+                    // TODO: Sleep
+                    sleep(CONNECTION_RETRY_TIMEOUT).await;
+                    continue;
+                }
+            };
+
+            info!(log, "Connected to peer"; "addr" => addr.to_string());
+
+            let (read_sock, write_sock, identify) = match perform_handshake(
+                sock,
+                &my_peer_id,
+                my_addr,
+                &log,
+            )
+            .await
+            {
+                Ok(val) => val,
+                Err(e) => {
+                    warn!(log, "Handshake error: {:?}", e; "addr" => addr.to_string());
+                    sleep(CONNECTION_RETRY_TIMEOUT).await;
+                    continue;
+                }
+            };
+
+            let log = log.new(o!("remote_peer_id" => identify.id.to_string()));
+
+            // Inform the main task that we have connected to a peer
+            let _ = main_tx
+                .send(ConnToMainMsg {
+                    handle_unique_id: unique_id,
+                    msg: ConnToMainMsgInner::ConnectedClient {
+                        addr: addr.clone(),
+                        peer_id: identify.id.clone(),
+                    },
+                })
+                .await;
+
+            let mut conn = EstablishedConn::new(
+                identify.id.clone(),
+                unique_id,
+                write_sock,
+                read_sock,
+                main_tx.clone(),
+                rx,
+                log.clone(),
+            );
+
+            // We can only get errors back from `conn.run`
+            match conn.run().await {
+                ConnErr::Retry => {
+                    // The only thing we need to keep is our receiver from main
+                    rx = conn.rx;
+                }
+                ConnErr::Close => {
+                    // The Main task told us to shutdown
+                    return;
+                }
+            }
+        }
+    });
+    PeerConnHandle { handle, tx, addr, unique_id }
+}
+
+/// Spawn a task that handles accepted connections from a peer
+pub async fn spawn_server(
+    unique_id: u64,
+    my_peer_id: Baseboard,
```

Comment:
Nit: this naming is weird to me, because if it's my ID, it can't belong to a peer (defined as someone _else_ that we're connecting to).  Maybe just `my_id`?